# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/chromium/src/chrome/common/ipc_channel.h
# Commit: 092bc4ac84dc
# Full Hash: 092bc4ac84dce8e4b5cd6e23377652596cf74204
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-23 09:53:24
# Regressor Bug: 1706374
# File Overlap Count: 1
# Description:
#   Bug 1706374 - Part 9: Allow reading OtherPid() from an opened IPC::Channel, r=handyman
#   
#   This will be used to handle pre-opened channels more reliably in ports code.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D112774
# ==============================================================================

diff -r 3b2722d8636b -r 092bc4ac84dc ipc/chromium/src/chrome/common/ipc_channel.h
--- a/ipc/chromium/src/chrome/common/ipc_channel.h	Tue Jun 22 18:17:20 2021 +0000
+++ b/ipc/chromium/src/chrome/common/ipc_channel.h	Tue Jun 22 18:17:21 2021 +0000
@@ -12,6 +12,7 @@
 #include "base/basictypes.h"
 #include "build/build_config.h"
 #include "mozilla/UniquePtr.h"
+#include "chrome/common/ipc_message.h"
 
 #ifdef OS_WIN
 #  include <string>
@@ -121,6 +122,10 @@
   // immediately.
   bool Send(mozilla::UniquePtr<Message> message);
 
+  // The PID which this channel has been opened with. This will be
+  // `-1` until `OnChannelConnected` has been called.
+  int32_t OtherPid() const;
+
   // Unsound_IsClosed() and Unsound_NumQueuedMessages() are safe to call from
   // any thread, but the value returned may be out of date, because we don't
   // use any synchronization when reading or writing it.